### DISCUSSION TITLE
When there is a single TF message for a particular frame, it "sticks" to parent frame instead of staying in position

### DISCUSSION THREAD

**Discussion by bernard-bear** [foxglove/community#875] [Q&A]
**Description**

When there is a single TF message for a particular frame, it "sticks to" the parent frame. This behaviour deviates from Rviz, where the frame stays in position.

- Version: 1.74.1+2c854ff.9701fac
- Platform: Web app. Bug is also reproduced in desktop app (Ubuntu)
- Data source (e.g. bag file, mcap file, rosbridge, ROS 1/2 native): bag file (I can provide the rosbag via direct message if needed)

**Steps To Reproduce**

See video:

[TF Bug Foxglove Studio.webm](https://github.com/foxglove/studio/assets/138452214/c40c8ff0-3897-4fde-b64c-84a70afb87d5)

1. Open rosbag containing a single TF message for a particular frame, with the parent frame as `base_link`.
2. Play the rosbag.
3. Observe that once `tag` appears, it "sticks to" the parent frame (`base_link`) and follows it.

**Expected Behavior**

I expect similar behaviour as Rviz, where `tag` does not follow the parent frame and stays in position (fading away eventually). See video:

[TF Bug Rviz.webm](https://github.com/foxglove/studio/assets/138452214/238a1fcf-8048-48eb-b4fe-24d42beaff51)

**Comment by foxhubber** [comment]
Internal tracking ticket: [FG-5412](https://linear.app/foxglove/issue/FG-5412/when-there-is-a-single-tf-message-for-a-particular-frame-it-sticks-to)

**Comment by achim-k** [comment]
Foxglove Studio currently does not distinguish between static and non-static transforms. All transforms are treated as static transforms, so there is no maximum lifetime for a transformation after which it expires. 

In ROS, transforms are considered static / non-static depending on which topic transforms are published on `/tf_static` and `/tf`, respectively. Studio would have to take that into account when adding transforms to it's internal tf buffer. 

Since this is a missing feature, I'm going ahead and remove the bug label and move it to a feature request.

**Comment by defunctzombie** [comment]
If I understand the video and what you want to accomplish, you can parent your `tag` transform against `map` and it will be stationary. This will work in Foxglove and Rviz and you won't rely on any topic-specific behavior.
